File: easy-leetcode-solutions/cpp-solutions/longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string res;
        for (int i = 0; i < strs[0].length(); ++i) {
            char c = strs[0][i];
            for (int k = 1; k < strs.size(); ++k) {
                if (i == strs[k].length() || strs[k][i] != c) return res;
            }
            res += c;
        }
        return res;
    }
};

File: easy-leetcode-solutions/python-solutions/longest-common-prefix.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        solution = ""
        for k in range(len(strs[0])):
            c = strs[0][k]
            for i in range(1, len(strs)):
                if k >= len(strs[i]) or strs[i][k] != c:
                    return solution
            solution += c
        return solution
